### PR TITLE
Added docs on AWS with OIDC.

### PR DESCRIPTION
**Pull request by SushrutHarness**
# Harness Developer Pull Request
This feature is behind this flag: `CDS_AWS_OIDC_AUTHENTICATION`

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [X] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md [hunk 1]
```diff
@@ -566,6 +572,9 @@ Create a Harness AWS Connector to connect to your AWS account using the IAM User
 3. In **AWS Cloud Provider**, in **Name**, enter a name, and click **Continue**.
 4. In Credentials, enter the following and click **Continue**:
    1. Select **AWS Access Key**.
+   :::note
+   If you would like to use AWS with OIDC, select **Use OIDC** here and ignore step 3 below.
```
Comment (SushrutHarness) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md:576]: Two questions:

Is this correct?
Does this bloat the docs?

File: docs/continuous-delivery/cd-integrations.md [hunk 2]
```diff
@@ -227,4 +227,8 @@ To enable a feature flag in your Harness account, contact [Harness Support](mail
         <td>CDS_ECS_BG_VALIDATION</td>
         <td>If green services exist in your Blue Green deployment, you can configure Harness to update those services instead of deleting them and then re-creating them with a new manifest and artifact.<br/>For more information, go to <a href="/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial/#update-green-services">Update green services</a> and <a href="/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial/#ecs-blue-green-service-validation">ECS blue/green service validations</a>.<br/><b>This feature is in Beta, and it requires Harness Delegate version 23.11.81820.</b> </td>
     </tr>
+    <tr>
+        <td>CDS_AWS_OIDC_AUTHENTICATION</td>
+        <td>Enables the option to connect to AWS with OIDC. Only currently supported for Kubernetes, Helm, Terraform, and Cloudformation. <b>This feature is in Beta.</b> </td>
```
Comment (SushrutHarness) [inline docs/continuous-delivery/cd-integrations.md:232]: i've listed this feature as in Beta. Is that correct @thisrohangupta ?

File: docs/platform/connectors/cloud-providers/add-aws-connector.md [hunk 3]
```diff
@@ -49,12 +49,18 @@ The AWS [IAM Policy Simulator](https://docs.aws.amazon.com/IAM/latest/UserGuide/
...
    * **AWS Access Key:** Provide an [Access Key and Secret Access Key](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) for the IAM role you want the connector to use.
    * **Use IRSA:** Required if you use IAM roles for service accounts, including IRSA in EKS clusters with OIDC Provider. Additional configuration is required, as explained in [Use IRSA](#use-irsa).
+   * **Use OIDC**: Select this option to connect to AWS with OIDC. This requires the user to create an OIDC identity provider in AWS and add it in a trust relationship with an IAM role that Harness will assume when doing any AWS operations with this connector.
+   :::note
+
+   The `Use OIDC` credential option is currently behind a feature flag. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/cloud-providers/add-aws-connector.md:61]: @SushrutHarness please mention the feature flag here.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 4]
```diff
@@ -422,6 +422,14 @@ Specify the credentials that enable Harness to connect your AWS account. There a
...
 * **Use IRSA:** Allows the Harness Kubernetes delegate in AWS EKS to use a specific IAM role when making authenticated requests to resources. By default, the Harness Kubernetes delegate uses a ClusterRoleBinding to the **default** service account; whereas, with this option, you can use AWS [IAM roles for service accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to associate a specific IAM role with the service account used by the Harness Kubernetes delegate. For instructions, go to [Use IRSA](/docs/platform/connectors/cloud-providers/add-aws-connector/#use-irsa).
+* **Use OIDC**: Connect to AWS with OIDC. In order to do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. 
+
+:::note
+
+The `Use OIDC` credential option is currently behind a feature flag. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:429]: mention the feature flag

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 5]
```diff
@@ -422,6 +422,14 @@ Specify the credentials that enable Harness to connect your AWS account. There a
...
+
+This option is only currently supported for Kubernetes, Helm, Terraform, and CloudFormation deployment types.
```
Comment (roshnisarangadharan) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:431]:
```suggestion
Currently, this option is supported only for Kubernetes, Helm, Terraform, and CloudFormation deployment types. 
```

File: docs/continuous-delivery/cd-integrations.md (hunk 2 above)
Comment (roshnisarangadharan) [inline docs/continuous-delivery/cd-integrations.md:232]:
```suggestion
        <td>Enables the option to connect to AWS with OIDC. Currently, this option is supported only for Kubernetes, Helm, Terraform, and CloudFormation deployment types. <b>This is a Beta feature.</b> </td>
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md [hunk 6]
```diff
@@ -95,6 +95,12 @@ When you do your own deployments, you might need additional permissions added to
 1. Create or edit an IAM user and attach the custom managed policy you create above. In this example, we'll name this policy **HarnessECS**.
 2. Generate an access key for the IAM user for connecting Harness to AWS ECS. In the IAM User, click **Security credentials** and then in **Access keys** create and save the key. You will use the Access key ID and Secret access key in the Harness AWS connector later.
 
+:::note 
+
+If you are using the OIDC credential type for your AWS connector, also add a trust relationship between this IAM role and your identity provider. For information on using OIDC with AWS at Harness see: [AWS Connector -- Credentials](/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md#credentials)
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md:100]:
```suggestion
If you are using the OIDC credential type for your AWS connector, also add a trust relationship between this IAM role and your identity provider. For information on using OIDC with AWS at Harness, go to [AWS Connector -- Credentials](/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md#credentials)
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/spot-deployment.md [hunk 7]
```diff
@@ -27,6 +28,12 @@ To connect to a Spot cloud provider using Spot's API, go to [Spot API authentica
 
 To use an AWS connector, your spot account must first be connected to the AWS cloud provider. For more information, go to [Connect your cloud account to Spot](https://docs.spot.io/connect-your-cloud-provider/aws-account).
 
+:::note
+
+Currently, Harness does not support using Spot Elastigroup with an OIDC-enabled AWS connector.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/spot-deployment.md:33]: This info is covered in line 21 already right? You can remove this note.

**Review by roshnisarangadharan — CHANGES_REQUESTED** [review]
@SushrutHarness added few editorial comments, amend as you see fit. Please mention the feature flag names.

File: docs/continuous-delivery/cd-integrations.md (hunk 2 above)
Comment (thisrohangupta) [reply docs/continuous-delivery/cd-integrations.md]: Yes this feature is in Beta. You need to add ECS to the list.

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md (hunk 1 above)
Comment (thisrohangupta) [reply docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md:576]: Use OIDC is another option this doesn't bloat the docs

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 8]
```diff
@@ -422,6 +422,14 @@ Specify the credentials that enable Harness to connect your AWS account. There a
...
+* **Use OIDC**: Connect to AWS with OIDC. In order to do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. 
+
+:::note
+
+The `Use OIDC` credential option is currently behind the feature flag `CDS_AWS_OIDC_AUTHENTICATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+Currently, this option is only supported for Kubernetes, Helm, Terraform, and CloudFormation deployment types.
```
Comment (SushrutHarness) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:431]:
```suggestion
Currently, this option is only supported for Kubernetes, Helm, Terraform, ECS, and CloudFormation deployment types. 
```

File: docs/continuous-delivery/cd-integrations.md [hunk 9]
```diff
@@ -227,4 +227,8 @@ To enable a feature flag in your Harness account, contact [Harness Support](mail
         <td>CDS_ECS_BG_VALIDATION</td>
         <td>If green services exist in your Blue Green deployment, you can configure Harness to update those services instead of deleting them and then re-creating them with a new manifest and artifact.<br/>For more information, go to <a href="/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial/#update-green-services">Update green services</a> and <a href="/docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial/#ecs-blue-green-service-validation">ECS blue/green service validations</a>.<br/><b>This feature is in Beta, and it requires Harness Delegate version 23.11.81820.</b> </td>
     </tr>
+    <tr>
+        <td>CDS_AWS_OIDC_AUTHENTICATION</td>
+        <td>Enables the option to connect to AWS with OIDC. Currently, this option is only supported for Kubernetes, Helm, Terraform, and Cloudformation. <b>This is a Beta feature.</b> </td>
```
Comment (SushrutHarness) [inline docs/continuous-delivery/cd-integrations.md:232]:
```suggestion
        <td>Enables the option to connect to AWS with OIDC. Currently, this option is only supported for Kubernetes, Helm, Terraform, ECS, and Cloudformation. <b>This is a Beta feature.</b> </td>
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md [hunk 10]
```diff
@@ -95,6 +95,12 @@ When you do your own deployments, you might need additional permissions added to
 1. Create or edit an IAM user and attach the custom managed policy you create above. In this example, we'll name this policy **HarnessECS**.
 2. Generate an access key for the IAM user for connecting Harness to AWS ECS. In the IAM User, click **Security credentials** and then in **Access keys** create and save the key. You will use the Access key ID and Secret access key in the Harness AWS connector later.
 
+:::note 
+
+If you are using the OIDC credential type for your AWS connector, also add a trust relationship between this IAM role and your identity provider. For information on using OIDC with AWS at Harness, go to [AWS Connector -- Credentials](/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md#credentials).
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md:100]:
```suggestion
If you are using the OIDC credential type for your AWS connector, add a trust relationship between the IAM role and your identity provider. For information on using OIDC with AWS at Harness, go to [AWS Connector -- Credentials](/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md#credentials).
```

File: docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md (hunk 1 above)
Comment (roshnisarangadharan) [inline docs/continuous-delivery/deploy-srv-diff-platforms/aws/ecs/ecs-deployment-tutorial.md:576]:
```suggestion
   If you would like to use AWS with OIDC, select **Use OIDC** here and ignore Step 3 below. 
```

**Review by roshnisarangadharan — APPROVED** [review]

**Pull request merged**